eos/scattering/ee-to-ccbar.cc

```diff
@@ -0,0 +1,756 @@
+                        break;
+                    case DstpDstmF2:
+                        channel_array[i] = std::make_shared<FWaveVVChannel<EEToCCBar::nchannels, EEToCCBar::nresonances, EEToCCBar::order>>(channel_names[i], m_Dst, m_Dst, _get_g0_column(_filter_channel_index(Channels(i)), std::make_index_sequence<EEToCCBar::nresonances>()));
+                        break;
+                    case DspDsstm:
+                    case DsstpDsm:
+                        channel_array[i] = std::make_shared<PWaveVPChannel<EEToCCBar::nchannels, EEToCCBar::nresonances, EEToCCBar::order>>(channel_names[i], m_Ds, m_Dsst, _get_g0_column(_filter_channel_index(Channels(i)), std::make_index_sequence<EEToCCBar::nresonances>()));
+                        break;
+                    case DsstpDsstmP0:
+                    case DsstpDsstmP2:
+                        channel_array[i] = std::make_shared<PWaveVVChannel<EEToCCBar::nchannels, EEToCCBar::nresonances, EEToCCBar::order>>(channel_names[i], m_Dsst, m_Dsst, _get_g0_column(_filter_channel_index(Channels(i)), std::make_index_sequence<EEToCCBar::nresonances>()));
+                        break;
+                    case DsstpDsstmF2:
+                        channel_array[i] = std::make_shared<FWaveVVChannel<EEToCCBar::nchannels, EEToCCBar::nresonances, EEToCCBar::order>>(channel_names[i], m_Dsst, m_Dsst, _get_g0_column(_filter_channel_index(Channels(i)), std::make_index_sequence<EEToCCBar::nresonances>()));
+                        break;
+
+                    default:
+                        throw InternalError("The number of requested channels (array) is larger than the number of known channel names.");
+                }
+            }
+
+            K = std::shared_ptr<KMatrix<EEToCCBar::nchannels, EEToCCBar::nresonances, EEToCCBar::order>> (
+                new KMatrix<EEToCCBar::nchannels, EEToCCBar::nresonances, EEToCCBar::order>(
+                    channel_array,
+                    resonance_array,
+                    _get_bkgpol(std::make_index_sequence<EEToCCBar::order + 1>(), std::make_index_sequence<EEToCCBar::nchannels>(), std::make_index_sequence<EEToCCBar::nchannels>()),
+                    "e^+e^-->ccbar"
+                    )
+                );
+        }
+
+        const IntermediateResult * prepare(const double & E)
+        {
+            _intermediate_result.tmatrix_row_0 = K->tmatrix_row(0, E*E);
+
+            _intermediate_result.E = E;
+            _intermediate_result.s = E*E;
+
+            return &_intermediate_result;
+        }
+
+
+        inline double sigma_eetomumu(const double & E)
+        {
+            // Conversion factor between GeV^2 and nb
+            const double speedoflight = 299792458.; // Exact value
+            const double GeVtonb = 10 * pow( 1.e18 * hbar * speedoflight, 2.0);
```

Comment:
why the "pow(x,2.0)" and not "power_of<2>(x)"?

Reply:
Mistake of mine, thanks!